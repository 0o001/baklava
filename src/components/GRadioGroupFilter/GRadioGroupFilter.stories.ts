import { storiesOf } from '@storybook/vue';
import GBox from '../GBox';
import GRadioGroupFilter from './GRadioGroupFilter.vue';

storiesOf('GRadioGroupFilter', module)
  .addParameters({ component: GRadioGroupFilter })
  .add('Default', () => ({
    components: { GBox, GRadioGroupFilter },
    data () {
      return {
        selectedBrand: 'samsung',
        brands: [{
          name: 'Apple',
          value: 'apple',
        }, {
          name: 'Samsung',
          value: 'samsung',
        }, {
          name: 'Huawei',
          value: 'huawei',
        }],
      };
    },
    template: `
      <GBox flex direction="row" class="g-m-10">
        <GRadioGroupFilter
          v-for="(brand, index) in brands"
          :key="index"
          :val="brand.value"
          name="brands"
          v-model="selectedBrand">
          {{ brand.name }}
        </GRadioGroupFilter>
      </GBox>
    `,
  }));
