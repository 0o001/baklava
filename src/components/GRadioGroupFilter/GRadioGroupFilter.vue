<template>
  <label
    class="g-radio-group-filter"
    :class="{
      '-disabled': $attrs.disabled,
      '-checked': isChecked
    }"
  >
    <input
      type="radio"
      v-bind="$attrs"
      v-on="listeners"
      :name="name"
      :value="val"
      :checked="isChecked"
      @click="click"
    >
    <span>
      <slot />
    </span>
  </label>
</template>

<script lang="ts">
export default {
  name: 'GRadioGroupFilter',
  components: {},
  props: {
    value: {
      type: [Boolean, String, Number, Object],
      default: '',
    },
    val: {
      type: [Boolean, String, Number, Object],
      default: '',
    },
    name: {
      type: String,
      default: '',
    },
  },
  methods: {
    click (): void {
      if (!this.$attrs.disabled && this.value !== this.val) {
        this.$emit('input', this.val);
      }
    },
  },
  computed: {
    isChecked (): boolean {
      return this.value === this.val;
    },
    listeners (): {} {
      const { input, ...rest } = this.$listeners;

      return rest;
    },
  },
};
</script>

<style lang="scss" scoped>
.g-radio-group-filter {
  cursor: pointer;
  border-radius: 16px;
  border: solid 1px var(--mid-grey-500);
  background-color: var(--white);
  padding: 8px 20px;
  font-size: var(--font-size-14);
  color: var(--mid-grey-500);

  input[type=radio] {
    width: 0;
    height: 0;
    opacity: 0;
    pointer-events: none;
  }

  span {
    line-height: normal;
  }

  &:not(:last-child) {
    margin-right: 10px;
  }

  &.-checked {
    background-color: var(--main-grey-500);
    border: none !important;
    color: var(--white);
  }
}
</style>
