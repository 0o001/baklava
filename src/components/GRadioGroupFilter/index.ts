import GRadioGroupFilter from './GRadioGroupFilter.vue';

export default GRadioGroupFilter;
