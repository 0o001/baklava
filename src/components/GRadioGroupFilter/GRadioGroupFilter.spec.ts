import { mount } from '@vue/test-utils';
import GRadioGroupFilter from './';

describe('GRadioGroupFilter', () => {
  let wrapper: any;

  const renderWrapper = (options?: any) => {
    wrapper = mount(GRadioGroupFilter, {
      propsData: {
        name: 'Apple',
        value: 'apple',
        val: 'apple',
      },
      ...options,
    });
  };

  beforeEach(() => {
    renderWrapper();
  });

  it('should match with snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should match with snapshot with unchecked', async () => {
    await wrapper.setProps({
      value: 'samsung',
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should add checked class if radio button is checked', () => {
    expect(wrapper.classes()).toContain('-checked');
  });

  it('should remove checked class if radio button is not checked', async () => {
    await wrapper.setProps({
      value: 'samsung',
    });

    expect(wrapper.classes('-checked')).toBe(false);
  });

  it('should emit selected value when clicked radio button', async () => {
    await wrapper.setProps({
      value: 'samsung',
    });

    await wrapper.trigger('click');

    expect(wrapper.emitted().input).toBeTruthy();
    expect(wrapper.emitted().input[0][0]).toEqual(wrapper.vm.val);
  });
});
