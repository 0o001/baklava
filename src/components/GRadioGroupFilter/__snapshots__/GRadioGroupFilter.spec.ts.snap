// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GRadioGroupFilter should match with snapshot 1`] = `
<label
  class="g-radio-group-filter -checked"
>
  <input
    name="Apple"
    type="radio"
    value="apple"
  />
   
  <span />
</label>
`;

exports[`GRadioGroupFilter should match with snapshot with unchecked 1`] = `
<label
  class="g-radio-group-filter"
>
  <input
    name="Apple"
    type="radio"
    value="apple"
  />
   
  <span />
</label>
`;
